launch/launch/actions/clear_launch_configurations.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for the ClearLaunchConfigurations action."""
+
+from typing import List
+from typing import Optional
+
+from ..action import Action
+from ..launch_context import LaunchContext
+from ..some_substitutions_type import SomeSubstitutionsType
+
+from ..utilities import perform_substitutions
+from ..utilities import normalize_to_list_of_substitutions
+
+
+class ClearLaunchConfigurations(Action):
+    """
+    Action that clears launch configurations unless explicitly excluded.
+
+    If launch_configurations_to_not_be_cleared is not set or an empty
+    list, all launch configurations are cleared from the current context.
+
+    If a member of launch_configurations_to_not_be_cleared matches a key
+    in the existing context, it will be preserved in the current context.
+    All other launch configurations will be cleared from the current.
+    context.
+    """
+
+    def __init__(
+        self,
+        launch_configurations_to_not_be_cleared: Optional[List[SomeSubstitutionsType]] = [],
```

Comment:
I would avoid defaulting to `[]`, and instead use `None`, due to a common python gotcha referred to as the "mutable default argument anti-pattern", see: https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

Reply:
Thanks for the link, that makes sense. I will make the change and add in tests.